internal/sandbox/tarfs/tarfs_test.go

```diff
@@ -40,3 +42,39 @@ func makeTarFS(t *testing.T, path string) sandbox.FileSystem {
 	assert.OK(t, err)
 	return fsys
 }
+
+func TestAlpine(t *testing.T) {
+	targz, err := os.Open("testdata/alpine.tar.gz")
+	assert.OK(t, err)
+	defer targz.Close()
+
+	tarball, err := os.Create(filepath.Join(t.TempDir(), "alpine.tar"))
+	assert.OK(t, err)
+	defer tarball.Close()
+
+	gunzip, err := gzip.NewReader(targz)
+	assert.OK(t, err)
+	defer gunzip.Close()
+
+	_, err = io.Copy(tarball, gunzip)
+	assert.OK(t, err)
+
+	_, err = tarball.Seek(0, 0)
+	assert.OK(t, err)
+
+	fsys, err := tarfs.OpenFile(tarball)
+	assert.OK(t, err)
+
+	f, err := fsys.Open("/usr/share/apk/keys/aarch64/alpine-devel@lists.alpinelinux.org-58199dcc.rsa.pub", sandbox.O_RDONLY, 0)
+	assert.OK(t, err)
+	defer f.Close()
+
+	s, err := f.Stat("", 0)
+	assert.OK(t, err)
+	assert.Equal(t, s.Size, 451)
```

Comment:
Should it check whether it matches an expected byte slice (or hash) too? It could be any 451 bytes from the tar file otherwise?

Reply:
Good call 👍